syncer/syncer.go

```diff
@@ -1094,7 +1094,7 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		if s.cfg.Mode == config.ModeAll {
 			if err = s.flushCheckPoints(); err != nil {
 				s.tctx.L().Warn("fail to flush checkpoints when starting task", zap.Error(err))
-			} else {
+			} else if s.cfg.CleanDumpFile {
```

Comment:
this may fix the error in #941?

Reply:
we default enabled `CleanDumpFile`, so if this line was added #941 still happens. Maybe we should check downstream checkpoint, if not exist then load from file 😢

Reply:
previous `if fresh` can achieve this?